src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -2717,7 +2717,7 @@ const Type* NeverBranchNode::Value(PhaseGVN* phase) const {
 //------------------------------Ideal------------------------------------------
 // Check for no longer being part of a loop
 Node *NeverBranchNode::Ideal(PhaseGVN *phase, bool can_reshape) {
-  if (can_reshape && !in(0)->is_Loop()) {
+  if (can_reshape && !in(0)->is_Region()) {
```

Comment:
I assume this is the actual fix.

Reply:
Yes, it is. The rest is shenandoah specific.

Reply:
okay